docs/architecture/blazor-for-web-forms-developers/migration.md

```diff
@@ -1,15 +1,646 @@
+    {
+        Configuration = configuration;
+        Env = env;
+    }
+
+    public IConfiguration Configuration { get; }
+
+    public IWebHostEnvironment Env { get; }
+
+    // This method gets called by the runtime. Use this method to add services to the container.
+    // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddRazorPages();
+        services.AddServerSideBlazor();
+
+        if (Configuration.GetValue<bool>("UseMockData"))
+        {
+            services.AddSingleton<ICatalogService, CatalogServiceMock>();
+        }
+        else
+        {
+            services.AddScoped<ICatalogService, CatalogService>();
+            services.AddScoped<IDatabaseInitializer<CatalogDBContext>, CatalogDBInitializer>();
+            services.AddSingleton<CatalogItemHiLoGenerator>();
+            services.AddScoped(_ => new CatalogDBContext(Configuration.GetConnectionString("CatalogDBContext")));
+        }
+    }
+
+    // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+    public void Configure(IApplicationBuilder app, ILoggerFactory loggerFactory)
+    {
+        loggerFactory.AddLog4Net("log4Net.xml");
+
+        if (Env.IsDevelopment())
+        {
+            app.UseDeveloperExceptionPage();
+        }
+        else
+        {
+            app.UseExceptionHandler("/Home/Error");
+        }
+
+        // Middleware for Application_BeginRequest
+        app.Use((ctx, next) =>
+        {
+            LogicalThreadContext.Properties["activityid"] = new ActivityIdHelper(ctx);
+            LogicalThreadContext.Properties["requestinfo"] = new WebRequestInfo(ctx);
+            return next();
+        });
+
+        app.UseStaticFiles();
+
+        app.UseRouting();
+
+        app.UseEndpoints(endpoints =>
+        {
+            endpoints.MapBlazorHub();
+            endpoints.MapFallbackToPage("/_Host");
+        });
+
+        ConfigDataBase(app);
+    }
+
+    private void ConfigDataBase(IApplicationBuilder app)
+    {
+        using (var scope = app.ApplicationServices.CreateScope())
+        {
+            var initializer = scope.ServiceProvider.GetService<IDatabaseInitializer<CatalogDBContext>>();
+
+            if (initializer != null)
+            {
+                Database.SetInitializer(initializer);
+            }
+        }
+    }
+}
+```
+
+One significant change you may notice from Web Forms is the prominence of DI. DI has been a guiding principle in the ASP.NET Core design. It supports customization of almost all aspects of the ASP.NET Core framework. There's even a built-in service provider that can be used for many scenarios. If more customization is required, it can be supported by the many community projects. For example, you can carry forward your third-party DI library investment.
```

Comment:
Before using *DI*, the term should be introduced. Something like

... is the prominence of dependency injection (DI).

Reply:
The acronym is introduced in the **Enable startup process** section.

Reply:
The acronym is introduced in the **Enable startup process** section.

Reply:
Sounds good. Must have missed that.